cumulus/pallets/parachain-system/src/parachain_inherent.rs

```diff
@@ -182,29 +190,51 @@ impl<Message: InboundMessage> AbridgedInboundMessagesCollection<Message> {
 		(&self.full_messages, &self.hashed_messages)
 	}
 
-	/// Check that the current collection contains as many full messages as possible.
+	/// Check that the current collection contains at least 1 full message if needed.
+	pub fn check_enough_messages_included_basic(&self, collection_name: &str) {
+		if self.hashed_messages.is_empty() {
+			return;
+		}
+
+		// Here we just check that there is at least 1 full message.
+		assert!(
+			self.full_messages.len() >= 1,
```

Comment:
So the case that already the first message is too big can never happen, right?

Reply:
Very good point ! 

With the current Polkadot and Kusama configs it won't happen. But with a misconfiguration it probably could. Anyway this check is only used for DMP and the problem there is that we don't know the max message size in order to do a proper check. This will be fixed when we migrate DMP to  `check_enough_messages_included_advanced` as well. But in order to do this we need to also forward the max dmp message size to the parachain in the metadata. It would be great if we could do that in a separate PR and merge this one as it is. WDYT ?